.gitmodules

```diff
@@ -1,12 +1,14 @@
 [submodule "nettle"]
 	path = firmware/nettle
-	url = ../nettle.git
+	url = https://github.com/nthdimtech/nettle.git
```

Comment:
I intentionally did not link to githib here so that if the sources were mirrored somewhere else everything would come self contained from the mirror so long as the relative paths were maintained.

Reply:
Hmm. that's something to be considered. But in practice I've forked signet-base, and I could not do a 'git modules update'. Because of the relative paths.

It think, just like with mini-gmp, and crosstools-ng, it is better to have a full URL. Just because you happen to host both repositories it is not a reason to do differently. See this for example: https://github.com/monero-project/monero/blob/master/.gitmodules

Reply:
My rule has been that if I have made changes then I will use relative paths so it only applies to nettle for now. I have been considering moving my source code hosting for awhile while I think a lot of projects are okay with a hard coded dependency on github.

Another way of looking at this is that it makes creating a fork with more divergence easier since you're not going to end up with lingering links to the original repo. That said the instructions should probably note somewhere that the project is a collection of repos. Two notable examples of projects with relative paths are QT: https://github.com/qt/qt5/blob/dev/.gitmodules and the Android Open Source Project.

Reply:
Ok, fair enough, I've changed it back to relative path.